### PR TITLE
DAOS-12677 test: Update dmg storage query commands to use json output

### PR DESCRIPTION
**Pull request by phender**
Updating the dmg storage query set of commands to produce json output to avoid the pitfalls of searching command text output.

Skip-unit-tests: true
Test-tag: CsumErrorLog DmgStorageQuery DiskFailureTest NvmeHealth VmdLedStatus DmgPoolQueryRanks DmgPoolQueryTest DmgServerSetLogmasksTest DmgStorageQuery DmgStorageScanSCMTest SSDSocketTest

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/checksum/csum_error_logging.py [hunk 1]
```diff
@@ -19,88 +18,56 @@ class CsumErrorLog(DaosCoreBase):
...
+            int: the number of checksum errors on the device
+        """
+        checksum_errs = 0
+        info = get_dmg_smd_info(self, dmg.storage_query_device_health, 'devices', uuid=device_id)
+        for devices in info.values():
+            for device in devices:
+                try:
+                    if device['uuid'] == device_id:
+                        checksum_errs = device['health']['checksum_errs']
+                        break
```
Comment (daltonbohning) [inline src/tests/ftest/checksum/csum_error_logging.py:39]: This `break` only breaks out of the inner `for` loop. Should it break out of the outer as well? If so, maybe a simple
```suggestion
                        return device['health']['checksum_errs']
```

File: src/tests/ftest/control/dmg_storage_query.py [hunk 2]
```diff
@@ -117,37 +129,28 @@ def test_dmg_storage_query_device_health(self):
...
-            cels_temp = int("".join(re.findall(r"\d+", info[2]))) - 273.15
-            if not 0.00 <= cels_temp <= 71.00:
-                temp_err.append("{}".format(cels_temp))
-        if temp_err:
-            self.fail("Bad temperature on SSDs: {}".format(",".join(temp_err)))
+        errors = []
+        device_info = get_storage_query_device_info(self, self.dmg, health=True)
+        for device in device_info:
+            self.log.info("Health Info for %s:", device['uuid'])
+            for key in sorted(device['health']):
```
Comment (daltonbohning) [inline src/tests/ftest/control/dmg_storage_query.py:136]: FYI that `sorted` works on `dict.items()`
```suggestion
            for key, val in sorted(device['health'].items()):
```

File: src/tests/ftest/nvme/health.py [hunk 3]
```diff
@@ -68,27 +69,37 @@ def test_monitor_for_large_pools(self):
...
+                    for device in devices:
+                        try:
+                            if device['uuid'] == device and device['dev_state'] == 'NORMAL':
+                                passed = True
+                        except KeyError as error:
+                            self.fail(
+                                "Error parsing dmg.storage_query_device_health() output: {}".format(
+                                    error))
+                if not passed:
                     self.fail("device {} on host {} is not NORMAL".format(device, host))
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/health.py:103]: pylint:
`undefined-loop-variable, Using possibly undefined loop variable 'device'`

File: src/tests/ftest/checksum/csum_error_logging.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/checksum/csum_error_logging.py:39]: Done.

File: src/tests/ftest/nvme/health.py (hunk 3 above)
Comment (phender) [reply src/tests/ftest/nvme/health.py:103]: Fixed.

**Review by daltonbohning — DISMISSED** [review]

**Comment by phender** [comment]
Created the following new tickets for the 2 deployment/disk_failure.py manual test failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11651/12/:

- https://daosio.atlassian.net/browse/DAOS-12942
- https://daosio.atlassian.net/browse/DAOS-12943

All other tests passed.

**Review by dinghwah — APPROVED** [review]

**Pull request merged**